src/subplots.ts

```diff
@@ -12,1990 +12,2136 @@ var alert_count = 0;
  * A class that inherits from PlotData and is specific for drawing PrimitiveGroups.
  */
 export class PlotContour extends PlotData {
-    plot_datas:any;
-    selected: boolean = true;
-    public constructor(public data:any,
-                       public width: number,
-                       public height: number,
-                       public buttons_ON: boolean,
-                       public X: number,
-                       public Y: number,
-                       public canvas_id: string,
-                       public is_in_multiplot = false) {
-      super(data, width, height, buttons_ON, 0, 0, canvas_id, is_in_multiplot);
-      if (!is_in_multiplot) {
-        var requirement = '0.6.0';
-        check_package_version(data['package_version'], requirement);
-      }
-      this.plot_datas = [];
-      this.type_ = 'primitivegroup';
-      var d = this.data;
-      if (d['type_'] == 'primitivegroup') {
-        var a = PrimitiveGroup.deserialize(d);
-        this.plot_datas.push(a);
-        let multiple_labels_index = -1;
-        for (let i=0; i<a.primitives.length; i++) {
-          let primitive = a.primitives[i];
-          if (primitive.type_ === 'multiplelabels') {
-            multiple_labels_index = i;
-          } else {
-            this.minX = Math.min(this.minX, primitive.minX);
-            this.maxX = Math.max(this.maxX, primitive.maxX);
-            this.minY = Math.min(this.minY, primitive.minY);
-            this.maxY = Math.max(this.maxY, primitive.maxY);
-            if (["contour", "circle", "wire", "point"].includes(primitive.type_)) {
-              this.color_to_plot_data[primitive.hidden_color] = primitive;
-            }
+  plot_datas: any;
+  selected: boolean = true;
+  public constructor(public data: any,
```

Comment:
[Wait for refactor] Be aware of [JS Indentation Conventions](https://standardjs.com/rules.html) here